### DISCUSSION TITLE
HOGWARTS: Form not starting as hidden on DOM

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#75] [Help Me!] [answered]
Ben Jarrett

I have searched, but have not found a ticket with this specific issue.

When loading and hard resetting the DOM, the form does not start as hidden on the DOM. 
My function that hides the form when I click the sort button works correctly, but the form is populating on the DOM by default.

The only errors that appear in my inspector are the normal favicon and source map errors that I always have. 

DevTools failed to load SourceMap: Could not load content for chrome-extension://fheoggkfdfchfphceeifdbepaooicaho/sourceMap/chrome/iframe_handler.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

DevTools failed to load SourceMap: Could not load content for chrome-extension://fheoggkfdfchfphceeifdbepaooicaho/sourceMap/chrome/content.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

favicon.ico:1 GET http://localhost:8081/favicon.ico 404 (Not Found)

Googled :
https://www.w3schools.com/howto/howto_js_toggle_hide_show.asp
https://formidableforms.com/knowledgebase/javascript-examples/
https://www.w3schools.com/jsref/prop_hidden_form.asp

I have tried rewriting the function as ``` function toggleHide = () => { ```
Instead of ``` function toggleHide () { ```
Which is the way that I have it

I have tried putting the form above my jumbotron div. 

I have tried re-writing the function about 5 times. 

https://github.com/BenJarrett/sorting-hat/blob/hide-sorting/main.js
https://github.com/BenJarrett/sorting-hat/blob/hide-sorting/index.html

Loom video showing my issue:
https://www.loom.com/share/feaaed6986764f56bbfbb2d828bfe465

I'm probably making some tiny mistake somewhere, but I cannot find it for the life of me!

**Comment by drteresavasquez** [comment, marked as answer]
On app load, you are going to need to start with an initial state, in the HTML you have to tell it what the initial state of the display property is. So on the element in the HTML add `style="display: none;"` that should do it. 

Also, take that onclick off the button and put it inside your js file once you get this working. You'll need to add a query selector on app load. We don't want to add inline JS to our html as a permanent solution.

**Reply by BenJarrett** [reply]
Okay, the first part makes sense. 

I'm struggling to determine where and how to go about adding the query selector. 
Would I add this query selector inside of ``` const buttonEvents = () => {
  document.querySelector('#sortBtn').addEventListener('click', handleButtonClick); 
} ``` 
Or do I need to create a function that holds this query selector in a different part of the scope? 

Hope these questions make sense and are on the right track.

**Reply by drteresavasquez** [reply]
What have you tried? Also reference the pies example.

**Reply by drteresavasquez** [reply]
One more thing. The click event is a new topic, so search the other tickets to see if there are others that help or start a new issue.

**Reply by BenJarrett** [reply]
Earlier, during my lunch, I tried adding it into my buttonEvents function that holds the method chain of the sortBtn. 
```const buttonEvents = () => {
  document.querySelector('#sortBtn').addEventListener('click', handleButtonClick);
  document.querySelector('#toggle').addEventListener('click', handleButtonClick);
}``` 

I'll have to go back and work on this/ reference the pies example when I get done with work today. 

Thank you!

**Reply by drteresavasquez** [reply]
That looks good. I'll be on tonight if you run into any issues with the selector.

**Reply by BenJarrett** [reply]
Got it. I'll definitely be there
